lib/host.js

```diff
@@ -305,7 +335,47 @@ HostConnectionPool.prototype.forceShutdown = function () {
   });
 };
 
+/**
+ * Closes all open connections and schedules reconnection
+ * @param {Number} delay
+ */
+HostConnectionPool.prototype.scheduleReconnection = function (delay) {
+  this._forceShutdown();
+  this.reconnectionTimeout = setTimeout(this._attemptReconnection.bind(this), delay);
+};
+
+/**
+ * Prevents reconnection timeout from triggering
+ */
+HostConnectionPool.prototype.clearReconnection = function () {
+  if (!this.reconnectionTimeout) {
+    return;
+  }
+  clearTimeout(this.reconnectionTimeout);
+  this.reconnectionTimeout = null;
+};
+
+HostConnectionPool.prototype._attemptReconnection = function () {
+  this.reconnectionTimeout = null;
+  var c = this._createConnection();
+  var self = this;
+  c.open(function (err) {
+    if (err) {
+      self.log('info', util.format('Reconnection attempt to host %s failed', self.address), err);
+      self.host.setDown();
+      return c.close();
+    }
+    self.host.setUp();
```

Comment:
It might be a nice optimization somewhere around here to schedule `_maybeCreatePool` so the pool pre-emptively starts growing to core connections, this way we may not need to wait as long for the pool to grow to core when borrowing a connection after a reconnection.   I think this would help the scenario @gwicke [described](https://datastax-oss.atlassian.net/browse/NODEJS-155?focusedCommentId=23416&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23416).